stanford-earth/stanford_earth_helper
# EARTH-69a Stanford Profile import via CAP and Drupal Migrate APIs

READY FOR REVIEW

Summary

Adds a stanford_earth_capx sub-module to import profiles from the CAP API and create new Drupal user accounts for them.

Needed By (Date)
End of sprint

Urgency
Somewhat urgent-ish, I think.

Note: includes a tweak to the hook_entity_delete function in the stanford_earth_events module to restrict it to its own bundle.

Steps to Test

1. Check out this branch EARTH-69a branch on stanford_earth/se3_blt 
2. Run "composer update" - note: this will use the 8.9.x-dev version of acquia/blt
3. Run "blt sync:refresh"
4. Check that a number of stanford_person fields are added to the user entity
5. Enable Stanford Earth CAPx module
6. Check that a Profiles migration group is created at admin/structure/migrate
7. Click "List Migrations" - you will see a number of Undefined Index notices because the CAP API does not include every source field for every record if they are empty. I have a patch to submit for the migrate_plus module's json.php file to fix this.
8. The module is currently set to use a small workgroup of 3 people. Check the importer shows 3 unprocessed records.
9. Click Execute, then Import.
10. Check if three new user accounts were created for ksharp, aaronc, and crswan with Stanford
